### PR TITLE
Always instantiate Float32Array in the cost array of findOptimalSegmentationInternal.

### PR DESCRIPTION
**Pull request by rniwa**

### CODE REVIEW

File: resources/perf.webkit.org/public/v3/bundled-scripts.js [hunk 1]
```diff
@@ -674,7 +674,7 @@ addMetric(metric){this._metrics.push(metric);}
 isHidden(){return this._hidden;}}
 if(typeof module!='undefined')
 module.exports.Test=Test;class Metric extends LabeledObject{constructor(id,object)
-{super(id,object);this._aggregatorName=object.aggregator;object.test.addMetric(this);this._test=object.test;this._platforms=[];const suffix=this.name().match('([A-z][a-z]+|FrameRate)$')[0];this._unit={'FrameRate':'fps','Runs':'/s','Time':'ms','Duration':'ms','Malloc':'B','Heap':'B','Allocations':'B','Size':'B','Score':'pt','Power':'W',}[suffix];}
+{super(id,object);this._aggregatorName=object.aggregator;object.test.addMetric(this);this._test=object.test;this._platforms=[];const suffix=this.name().match('([A-Z][a-z]+|FrameRate)$')[0];this._unit={'FrameRate':'fps','Runs':'/s','Time':'ms','Duration':'ms','Malloc':'B','Heap':'B','Allocations':'B','Size':'B','Score':'pt','Power':'W',}[suffix];}
```
Comment (rniwa) [inline resources/perf.webkit.org/public/v3/bundled-scripts.js:677]: Huh, this is a weird typo in the built script. I wonder how that happened?

**Comment by rniwa** [comment]
## Chrome 120.0.6099.199

**Before**
29.3 ± 1.0
26.4 ± 1.4
27.4 ± 1.4

**After**
27.5 ± 2.4
28.6 ± 1.6
27.9 ± 1.0

## Firefox 121.0.1

**Before**
21.9 ± 1.7
22.0 ± 1.6
21.7 ± 1.7

**After**
22.4 ± 1.9
22.1 ± 1.9
21.8 ± 1.5


## Safari 17.2

**Before**
23.3 ± 0.90
23.8 ± 1.3
23.5 ± 1.4

**After**
24.8 ± 0.70
24.1 ± 1.3
24.3 ± 0.76

File: resources/perf.webkit.org/public/v3/bundled-scripts.js (hunk 1 above)
Comment (camillobruni) [reply resources/perf.webkit.org/public/v3/bundled-scripts.js:677]: As discussed in the meeting, this was likely from not regenerating the output after fixing it in #282 .

**Review by camillobruni — APPROVED** [review]

**Comment by julienw** [comment]
Do you have more context about this change?

**Comment by rniwa** [comment]
> Do you have more context about this change?
> 
> My understanding is that it's a fairly trivial change, please tell me if there's anything more than consistency.

The context is that `cost` array is always intended to be an array of `Float32Array` but we were forgetting to instantiate `Float32Array` for the very first item in the array. This PR makes it more consistent across the array.

**Comment by julienw** [comment]
Ok so it's clearly trivial

**Review by julienw — APPROVED** [review]

**Comment by jrmuizel** [comment]
@rniwa can you fix the same problem in JetStream too? 
https://github.com/WebKit/WebKit/blob/c6d8b99f59c25be18018634dc25fff0d66022287/PerformanceTests/JetStream2/worker/segmentation.js#L395

**Comment by rniwa** [comment]
> @rniwa can you fix the same problem in JetStream too? https://github.com/WebKit/WebKit/blob/c6d8b99f59c25be18018634dc25fff0d66022287/PerformanceTests/JetStream2/worker/segmentation.js#L395

Hm... I guess that would probably mean releasing JetStream 2.2 since 2.1 is already out? I'd create a PR but I'd leave it to folks maintaining JetStream to weigh in.

**Comment by rniwa** [comment]
Made a PR for JetStream 2 at https://github.com/WebKit/WebKit/pull/22688

**Pull request merged**